src/Containers/FileUpload/reducer.ts

```diff
@@ -0,0 +1,348 @@
+                            position: true,
+                        };
+                    return value;
+                }),
+            };
+        case IS_SUCCESS_RESTORE:
+            return {
+                ...state,
+                isSuccess: state.isSuccess.map((value, index) => {
+                    if (index === action.index)
+                        return {
+                            ...value,
+                            opacity: 1,
+                            position: false,
+                        };
+                    return value;
+                }),
+            };
+        case IS_FAILURE_FADE_OUT:
+            return {
+                ...state,
+                isFailure: state.isFailure.map((value, index) => {
+                    if (index === action.index)
+                        return {
+                            ...value,
+                            opacity: 0,
+                            position: true,
+                        };
+                    return value;
+                }),
+            };
+        case IS_FAILURE_RESTORE:
+            return {
+                ...state,
+                isFailure: state.isFailure.map((value, index) => {
+                    if (index === action.index)
+                        return {
+                            ...value,
+                            opacity: 1,
+                            position: false,
+                        };
+                    return value;
+                }),
+            };
+        case SET_INITIAL_HEIGHT_VALUES:
+            return {
+                ...state,
+                maxHeight: action.value,
+                totalHeight: action.value,
+            };
+        case SET_INITIAL_HEIGHT_PLUS_VALUES:
+            return {
+                ...state,
+                height: undefined,
+                maxHeight: MAX_HEIGHT,
+            };
+        case ADD_LOADING_IS_SUCCESS_IS_FAILURE:
+            return {
+                ...state,
+                loading:[...state.loading,{position:action.value.position,opacity:action.value.opacity}],
+                isSuccess:[...state.isSuccess,{position:action.value.position,opacity:action.value.opacity}],
+                isFailure:[...state.isFailure,{position:action.value.position,opacity:action.value.opacity}]
+            };
+        case RESET_LOADING_IS_SUCCESS_IS_FAILURE:
+            return{
+                ...state,
+                loading:[],
+                isSuccess:[],
+                isFailure:[],
+            };
+        case REMOVE_LOADING_IS_SUCCESS_IS_FAILURE:
+            return {
+                ...state,
+                loading:state.loading.filter((_,index)=>{
+                    if(index===action.index)return false
+                    return true
+                }),
+                isSuccess:state.isSuccess.filter((_,index)=>{
+                    if(index===action.index)return false
+                    return true
```

Comment:
Remove duplicate code @roggc

Reply:
Also why not

return A == B

Reply:
no, in fact is A!==B, I only want to remove one element, the one that gives A===B

Reply:
is A!==B, because I want all elements except the one for which A===B